src/stream/src/task/barrier_manager.rs

```diff
@@ -184,14 +185,15 @@ impl LocalBarrierManager {
     }
 
     /// Use `prev_epoch` to remove collect rx and return rx.
-    pub fn remove_collect_rx(&mut self, prev_epoch: u64) -> CompleteReceiver {
+    pub fn remove_collect_rx(&mut self, prev_epoch: u64) -> StreamResult<CompleteReceiver> {
         self.collect_complete_receiver
             .remove(&prev_epoch)
-            .unwrap_or_else(|| {
-                panic!(
+            .ok_or_else(|| {
```

Comment:
How about adding some comments (or simply link to this PR/issue) for why it's possible to fail here?